src/components/Table/Table.tsx

```diff
@@ -119,43 +125,144 @@ const Tbody = styled.tbody`
     gap: 0.25rem;
   }
 `;
+
+const SelectData = styled.td`
+  overflow: hidden;
+  ${({ theme }) => `
+    color: ${theme.click.table.row.color.text.default};
+    font: ${theme.click.table.cell.text.default};
+    padding: ${theme.click.table.body.cell.space.md.y} ${theme.click.table.body.cell.space.md.x};
+  `}
+  @media (max-width: 768px) {
+    width: auto;
+    align-self: stretch;
+    position: absolute;
+    left: 0;
+    top: 0;
+    bottom: 0;
+    ${({ theme }) => `
+      padding: ${theme.click.table.body.cell.space.sm.y} ${theme.click.table.body.cell.space.sm.x};
+      border-right: 1px solid ${theme.click.table.row.color.stroke.default};
+    `}
+  }
+`;
+
+const TableWrapper = styled.div`
+  width: 100%;
+  height: 100%;
+  overflow: auto;
+`;
+
+const TableOuterContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 1rem;
+`;
+
+const MobileActions = styled.div`
+  display: none;
+  @media (max-width: 768px) {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    padding: 0 ${({ theme }) => theme.click.table.body.cell.space.sm.x};
+  }
+`;
+
 interface TableCellType extends HTMLAttributes<HTMLTableCellElement> {
   label: ReactNode;
 }
-type TableRowType = Array<TableCellType>;
-interface TableProps extends HTMLAttributes<HTMLTableElement> {
+interface TableRowType extends HTMLAttributes<HTMLTableRowElement> {
+  checked?: boolean;
+  cells: Array<TableCellType>;
+}
+
+interface CommonTableProps
+  extends Omit<HTMLAttributes<HTMLTableElement>, "children" | "onSelect"> {
   headers: Array<TableHeaderProps>;
   rows: Array<TableRowType>;
 }
+
+interface SelectionType extends CommonTableProps {
+  selectable?: boolean;
+  onSelect: (index: "all" | number, checked: boolean) => void;
```

Comment:
how would this work out with selection and sorting? i think we should work with ids. we either require the `id` as a prop or we make sure that we can calculate a unique id for each row. Requiring an `id` prop sounds better to me right now

Reply:
Got it
Since we were not sure about the id I added the index
For select all should I still persist this behaviour?

Reply:
we can have a 

```js
onSelect:(ids: Array<string>) => void
```

function where we pass the list of selected ids

Reply:
can't we bubble up the item + the idx of it? I feel like we are going to do a lot of lookups per id on the client of the api, while we could easily bubble up the item and let client get whatever they want.

and api like `onSelect<T>(items: Array<{ item: T, idx: number })`

Reply:
the point is that the index might be difficult to work with because an item might change its index multiple times

Reply:
if the collection is coming from the outside, the index probably can be preserved. My main point is that we should return the entire object. Id will probably be in the object.

Reply:
Added the option

Reply:
I don't see any specific reason to pass the entire object, but if you do i'm fine with that